### ISSUE TITLE
Can't add otel-swift package in fresh project in xcode 13.3

### ISSUE THREAD

**Issue by johnbley** [open-telemetry/opentelemetry-swift#296]
Hi -
Using XCode 13.3 (13E113), I create a new ios app project and add opentelemetry-swift to it (up to next major version).  It fails with the message `the Swift tools version specification is missing a label; please consider inserting 'swift-tools-version:'...`. 

So, I look through all the `Package.swift` files in all the dependencies and they all declare the `swift-tools-version` just like they should.  I then iterate through all the otel-swift dependencies and try to add them to my empty project, one at a time (not actually adding any, just seeing if it would let me).  `Thrift-Swift` has the same behavior (can't add package, with an error message about swift-tools-version), and none of the other dependencies do.  `Thrift-Swift`'s `Package.swift` is very simple and seems correct.  

This is completely reproducible for me and the upstream behavior (can't add a project which depends on otel-swift, with the same error message) has been reproduced by several other folks at my company.  One reported that XCode 13.2 doesn't exhibit the problem, while 13.3 does.

Any thoughts?  Is this reproducible for anybody else in this community?

**Comment by nachoBonafonte** [comment]
Yes, I can see the same behaviour with opentelemetry-swift project, and as reported Xcode 13.2.1 does not exhibit the problem. But I cannot reproduce it with `Thrift-Swift`. The strange thing is that it works with the own opentelemetry-swift package and Xcode 13.3. I have no clue what it could be failing, it looks like a Xcode 13.3 bug

**Comment by cswelin** [comment]
I just tested with our main repo and an empty project both exhibit the error on Xcode 13.3.1:

```
the Swift tools version specification is missing a label; consider inserting 'swift-tools-version:' between the comment marker and the version specifier
```

**Comment by nachoBonafonte** [comment]
I have been testing it more in my projects, and continue working as expected for me, could it be a problem with the SPM cache in your system, so in 13.3.1 keeps the package cache from 13.3, or something like that?

**Comment by timorusky** [comment]
I am expiring the same issue on Xcode 13.3.1 (13E500a), first time I am importing this package. I think i found the issue, the [Thrift-Swift](https://github.com/undefinedlabs/Thrift-Swift) dependency contains the Tag `1.06` which corresponds to the commit [0c7f264](https://github.com/undefinedlabs/Thrift-Swift/commit/0c7f264ca773990af62b37d50c36c4e41bb1b10d). God knows why, but the SPM resolves this tag to the semver equivalent of `1.6.0`, therefore it tries to compute `0c7f264` and this commit contains a `Package.swift` that is missing the `swift-tools-version` annotation. I did some testing with a fork and deleting the tag `1.06` resolved the issue for me.

**Comment by nachoBonafonte** [comment]
Ok, I have deleted the tag from the referenced [Thrift-Swift](https://github.com/undefinedlabs/Thrift-Swift) dependency, lets hope it will fix the issues for everyone. Thanks a lot for your insight!!!

**Comment by timorusky** [comment]
Thank you! It is now working again for me. I was able to fetch the package 🎉

**Comment by johnbley** [comment]
Yep, this seems to have done it.  Thanks so much @timorusky!  From a debugging standpoint, did you find this by paying attention to the downloaded repo in `DerivedData` or by going through tags and/or git history of the `Package.swift`?

**Comment by timorusky** [comment]
@johnbley I was wondering what version the SPM was selecting, since the `Package.swift` of the latest version was obviously valid. Unfortunately I wasn't able to get that information from the repos in the build directory because as soon as SPM detected the invalid `Package.swift` it stopped the checkout. I built the SPM from source and debugged the [getBestAvailableVersion](https://github.com/apple/swift-package-manager/blob/6647fa09d6042b29cbb115eecf6f292beb7f6837/Sources/PackageGraph/Pubgrub/PubgrubDependencyResolver.swift#L1087) method, noticed that it resolved to `1.6.0` and figured out that it must be the malformed tag. I am sure there is an easier way tho.
